numpy/_core/_add_newdocs.py

```diff
@@ -810,7 +810,9 @@
         If true (default), then the object is copied.  Otherwise, a copy will
         only be made if ``__array__`` returns a copy, if obj is a nested
         sequence, or if a copy is needed to satisfy any of the other
-        requirements (``dtype``, ``order``, etc.).
+        requirements (``dtype``, ``order``, etc.). Note that any copy made is
+        shallow, which is mainly important for arrays containing Python
+        objects. See Notes from `copy`.
```

Comment:
The `Notes` of `np.copy` don't mention anything about this. It's only in the `Examples`. I'd probably move or copy those examples over to `ndarray.copy` as the canonical API for copying (`np.copy` refers to `ndarray.copy` as the preferred method).

I see that you copied the verbiage from `np.copy`, but I think that wording is misleading. I would dispute the idea that it's always a "shallow copy" that is "mainly important for arrays containing Python objects". If it's not an array that contains Python objects, the distinction between deep and shallow doesn't even exist. That wording raises a question what a "shallow copy" even means for the other dtypes. I'd prefer to flip it around: "For arrays containing Python objects (e.g. ``dtype=object``), the copy is a shallow one."